src/core/theme/utils/defaultTheme.ts

```diff
@@ -1,12 +1,21 @@
 import { suomifiTheme, ThemeProp } from '../';
+import { asPropType } from '../../../utils/typescript';
 
+/**
+ * Do defaults for component props
+ * Add default theme if theme not defined,
+ * Remove as-prop and change it to asProp if defined
+ * @param props Component props
+ */
 export const withDefaultTheme = <
```

Comment:
Typings for the function:
- `T extends { theme: ThemeProp; asProp?: asPropType }` : `T` contains theme and optional asProp
- `Partial<T> & { as?: asPropType }` : given function gets object that has `T` (all optional keys) and optional `as`
- that function returns `T`(containing theme and optional asProp as said earlier)